benchmark/attri_util.py

```diff
@@ -0,0 +1,132 @@
+import operator
+from dataclasses import dataclass
+from enum import Enum
+from functools import reduce
+from typing import Any, List, Optional, Tuple
+
+
+class ReadOnly:
+    def __init__(self, value):
+        self._value = value
+
+    @property
+    def value(self):
+        return self._value
```

Comment:
This structure is intended to protect data from unexpected changes. Since Python uses reference types by default rather than value types, it is susceptible to unintended modifications.